### DISCUSSION TITLE
How to turn off Thousand Separator (comma) for numbers, such as Year?

### DISCUSSION THREAD

**Discussion by KrishnaPG** [perspective-dev/perspective#2020] [Q&A]
The viewer table is adding thousand separators (commas) automatically to all number fields. But in some cases, such as the `Year` field the commas should not be added.

Please inform how to control the visibility of comma for some fields (columns) in the table?
![image](https://user-images.githubusercontent.com/4610831/204184394-aada13db-2aee-48df-bbf8-58fdb96f7b53.png)

**Comment by texodus** [comment]
Try casting to a date or string.  You can clean the data before loading it, or compute new cast columns using the expression feature:

```
viewer.restore({expressions: ["date(\"Fiscal_Year\")"]});
```

or

```
viewer.restore({expressions: ["string(\"Fiscal_Year\")"]});
```

Please refer to the [dedicated docs on expression columns](https://perspective.finos.org/docs/expressions).

**Reply by KrishnaPG** [reply]
Thank you @texodus .  Converting the type is certainly a work around, but changing the type introduces side effects such as different comparison operators, formatting options etc. 

Rather it would be nice to control the "presentation" of the number without changing the type. For example, it looks like `string` types are already allowing few format options (shown below). Similary it would be great to expose the format options for numbers also to control whether thousand separators should be applied or not for a particular field.

![image](https://user-images.githubusercontent.com/4610831/204192013-664cca2c-2b45-43fb-906e-f86c1bf1dc1d.png)

**Reply by KrishnaPG** [reply]
Added this as feature request, just in case relevant: https://github.com/finos/perspective/issues/2021

**Comment by texodus** [comment]
Not a workaround, this is the correct way to achieve this.  Presentation of `number` is defined by your browser's locale.

> ... but changing the type introduces side effects such as different comparison operators, formatting options etc.

This is not correct.  Nothing prevents you from applying the filter options on the original number column - filters don't care which column you filter on.
